src/selectors/dataset.js

```diff
@@ -14,16 +14,91 @@ export const datasetSelector = (config, state, instance) => {
   const baseState = config.baseSelector(state)
 
   // Get instance array
-  const instanceData = baseState.getIn(['instances', instance, 'data'])
+  let instanceData = baseState.getIn(['instances', instance, 'data'])
   const rawData = baseState.get('raw')
 
   // If instanceData is not created yet, return empty Map
   if(!instanceData) {
     return Map({})
   }
 
+  // Remove the pending.create items from the instance
+  instanceData = instanceData.filter(uid => baseState.getIn(['pending', 'create'], List()).indexOf(uid) === -1)
+
+  // Re-compute data
+  const computedData = instanceData.map(uid => {
+    let item = rawData.get(uid)
+
+    item = item.withMutations(map => {
+      // Check pending.update and add the pendingUpdate prop if necessary
+      const pendingUpdate = baseState.getIn(['pending', 'update'], List()).indexOf(uid) > -1
+      if (pendingUpdate) {
+        map.set('pendingUpdate', true)
+      }
+      // Check pending.remove and add the pendingRemove prop if necessary
+      const pendingRemove = baseState.getIn(['pending', 'remove'], List()).indexOf(uid) > -1
+      if (pendingRemove) {
+        map.set('pendingRemove', true)
+      }
```

Comment:
Get `baseState.getIn(['pending', 'update'], List())` and `baseState.getIn(['pending', 'remove'], List())` outside of loop.